packages/components/src/styles/checkbox-radio-styles.ts

```diff
@@ -67,125 +55,78 @@ export const getCheckboxRadioJssStyle = (
   const checkedHoverColorDark = formStateHoverColorDark || contrastHighColorDark;
 
   return {
-    '@global': {
-      ':host': addImportantToEachRule({
-        ...colorSchemeStyles,
-        ...hostHiddenStyles,
+    '::slotted': {
+      '&(input)': {
+        position: 'relative', // TODO: can be removed as soon as focus style was adjusted
+        width: fontLineHeight,
+        height: fontLineHeight,
+        font: `${fontSizeTextSmall} ${fontFamily}`, // needed for correct width and height definition based on ex-unit
         display: 'block',
-      }),
-      '::slotted': addImportantToEachRule({
-        '&(input)': {
-          gridArea: '1 / 1',
-          position: 'relative',
-          width: fontLineHeight,
-          height: fontLineHeight,
-          fontFamily, // needed for correct width and height definition
-          fontSize: '1rem', // needed for correct width and height definition
-          flexShrink: 0,
-          display: 'block',
-          margin: 0,
-          padding: 0,
-          WebkitAppearance: 'none', // iOS safari
-          appearance: 'none',
-          boxSizing: 'content-box',
-          background: `transparent 0% 0% / ${fontLineHeight}`,
-          transition: ['border-color', 'background-color'].map(getTransition).join(),
-          border: `2px solid ${uncheckedColor}`,
-          ...prefersColorSchemeDarkMediaQuery(theme, {
-            borderColor: uncheckedColorDark,
+        margin: 0,
+        padding: 0,
+        WebkitAppearance: 'none', // iOS safari
+        appearance: 'none',
+        boxSizing: 'content-box',
+        background: `transparent 0% 0% / ${fontLineHeight}`,
+        transition: ['border-color', 'background-color'].map(getTransition).join(),
+        border: `2px solid ${uncheckedColor}`,
+        outline: 0,
+        ...(disabledOrLoading
+          ? {
+              pointerEvents: 'none', // to prevent form element becomes clickable/toggleable
+            }
+          : {
+              cursor: 'pointer',
+            }),
+        ...prefersColorSchemeDarkMediaQuery(theme, {
+          borderColor: uncheckedColorDark,
+        }),
+      },
+      '&(input:checked)': {
+        // background-image is merged in later
+        borderColor: checkedColor,
+        backgroundColor: checkedColor,
+        ...prefersColorSchemeDarkMediaQuery(theme, {
+          borderColor: checkedColorDark,
+          backgroundColor: checkedColorDark,
+        }),
+      },
+      ...(!disabledOrLoading &&
+        !isHighContrastMode &&
+        hoverMediaQuery({
+          '&(input:hover),label:hover~* &(input)': {
```

Comment:
Would be good to comment what the `~*` is supposed to select.

Reply:
I've refactored it to `~.wrapper`, might be anyway better.